lib/utils/npm-utils.js

```diff
@@ -79,8 +97,47 @@ function fetchPeerDependencies(packageName) {
 
     if (error && error.code === "ENOENT") {
 
-        // TODO: should throw an error instead of returning null
-        return null;
+        // Fallback to using the npm registry API directly when npm is not available.
+        const { name, version } = parsePackageName(packageName);
+
+        try {
+            const data = await new Promise((resolve, reject) => {
+                https.get(`https://registry.npmjs.org/${name}`, res => {
```

Comment:
can we just use `node:fetch` - it's available in node.js >= 18?

Reply:
being a fallback, it's fine to use it - you can just use a `eslint-disable` comment.